sycl/test-e2e/lit.cfg.py

```diff
@@ -684,11 +688,17 @@
     sg_sizes = set(dev_sg_sizes[0]).intersection(*dev_sg_sizes)
     lit_config.note("SG sizes for {}: {}".format(sycl_device, ", ".join(sg_sizes)))
 
+    lit_config.note("Architectures for {}: {}".format(sycl_device, ", ".join(architectures)))
+    if len(architectures) != 1 or "unknown" in architectures:
+        architectures = set()
```

Comment:
I think it would be safer to just raise an error and abandon testing.

Reply:
If we see more (or less) than one architecture returned, we should indeed abandon testing and report that as some fatal failure.

However, if we see `unknown` we should proceed, I think. This is important for our downstream where we launch those tests on new/unstable/unknown hardware and we don't really want to have a complete blocker like this in there. It is ok that some of the tests may be skipped because they require a certain architecture, but at least the rest of the testing will go through.

Reply:
Can we require an explicit option telling us that "unknown" is fine? What I want to avoid is this auto-detection silently failing and then not running tests that we should be running on a particular runner/HW and thinking we have good test coverage there due to not seeing any failures when in reality the tests are simply skipped.

Reply:
> What I want to avoid is this auto-detection silently failing and then not running tests that we should be running on a particular runner/HW and thinking we have good test coverage there due to not seeing any failures when in reality the tests are simply skipped.

We could have a test for the detection which will fail indicating us that. And accepting `unknown` could be done as a downstream customization.

Anyways, the opt-in to allow `unknown` looks good to me as well. I think it is up to you, @aelovikov-intel, I'm not a code owner in this area.